src/daemon/Display.cpp

```diff
@@ -50,7 +54,30 @@
 #include "WaylandDisplayServer.h"
 
 
-namespace SDDM {
+namespace SDDM
+{
+    bool approveTty(const QString &desiredTty) {
+        if (Logind::isAvailable()) {
+            OrgFreedesktopLogin1ManagerInterface manager(Logind::serviceName(), Logind::managerPath(), QDBusConnection::systemBus());
+            auto reply = manager.ListSessions();
+            reply.waitForFinished();
+
+            const auto info = reply.value();
+            for(const SessionInfo &s : reply.value()) {
+                OrgFreedesktopLogin1SessionInterface session(Logind::serviceName(), s.sessionPath.path(), QDBusConnection::systemBus());
+                if (desiredTty == session.tTY()) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    int fetchAvailableVt() {
+        const auto vt = VirtualTerminal::currentVt();
+        return approveTty(QStringLiteral("tty%1").arg(vt)) ? vt : VirtualTerminal::setUpNewVt();
```

Comment:
On BSD:

currentVt will be -1
approveTTY will return true as we have no logind

so we return -1.  Same for if currentVt fails on Linux.

We want:
```
const auto vt = VirtualTerminal::currentVt();
if (vt > 0) {
    if (approveVt(vt) {
        return vt;
    }
} 
return setupNewVt();
```

Reply:
good catch, I wonder why it worked on the testing then... :thinking: